### PR TITLE
M3 Add maxmind data purge command

### PR DESCRIPTION
**Pull request by hluchas**
| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | https://github.com/mautic/mautic-documentation/pull/151
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | https://blog.maxmind.com/tag/ccpa/
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds a new command that will use the MaxMind Do Not Sell List to find data that we need to purge from our contacts and remove it.

#### Steps to test this PR:
1. Follow the testing steps for the command that Lukas added here: https://github.com/mautic/mautic/pull/8647. Running this command will get you the MaxMind Do not sell list.
2. Run this command `bin/console mautic:max-mind:purge`. It should complete successfully.
3. If there were not any matches found, then edit / add one of the IPs from the list into your ip_address table
4. Run again and see that a match was found.
5. To test purge, add a city to the details column for the ip you added in the ip_address table
```
insert into mautic_ip_addresses values (null,'73.189.157.100', 'a:2:{s:4:"city";s:6:"Boston";s:7:"country";s:2:"us";}');
```
6. Update the lead that is mapped to that IP to have the same city
7. Run the command again
8. The command should find a match, finish successfully and checking the contact will show that the city is now NULL

### CODE REVIEW

**Comment by hluchas** [comment]
Blocked by https://github.com/mautic/mautic/pull/8647
Build should fail now until this is merged

**Comment by RCheesley** [comment]
Bumping to 3.2 due to blocking PR pending feedback therefore this one cannot be merged.

**Comment by RCheesley** [comment]
Bumping to 3.2 due to PR blocking having unresolved conflicts.

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by hluchas** [comment]
Hi @RCheesley.
CLA log says 
```
2020-09-11T13:24:15.300Z INFO The contributors @mtshaw3 have not signed the CLA, adding error status to the pull request
```
He's no longer part of Acquia, neither Mautic contributor.

What we can do with this?

**Comment by RCheesley** [comment]
Looks like that's one that I missed on the CLA bot list, I'll take a look.

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@hluchas all sorted! 🎉

**Comment by hluchas** [comment]
Thank you @RCheesley  ;-)

File: app/bundles/CoreBundle/Tests/Unit/Command/MaxMindDoNotSellPurgeCommandTest.php
Comment (RCheesley) [inline app/bundles/CoreBundle/Tests/Unit/Command/MaxMindDoNotSellPurgeCommandTest.php]:
```suggestion
    protected function setUp(): void
```

Fix for PHPUnit 8

File: app/bundles/CoreBundle/Tests/Unit/IpLookup/MaxMindDoNotSellListTest.php
Comment (RCheesley) [inline app/bundles/CoreBundle/Tests/Unit/IpLookup/MaxMindDoNotSellListTest.php]:
```suggestion
    protected function setUp(): void
```
Fix for PHPUnit 8

**Review by RCheesley — CHANGES_REQUESTED** [review]
Tests need small fix for PHPUnit 8

**Comment by hluchas** [comment]
@RCheesley Looks good! But not the test coverage

**Comment by RCheesley** [comment]
> @RCheesley Looks good! But not the test coverage

That's probably because we just made a release so Staging will have some changes!

**Review by escopecz — APPROVED** [review]
Looks good 👍

**Review by RCheesley — APPROVED** [review]

**Pull request merged**
